options.h build warnings

```
../Source/options.h: In constructor ‘devilution::OptionEntryBase::OptionEntryBase(devilution::string_view, devilution::OptionEntryFlags, devilution::string_view, devilution::string_view)’:
../Source/options.h:70:26: warning: ‘devilution::OptionEntryBase::flags’ will be initialized after [-Wreorder]
   70 |         OptionEntryFlags flags;
      |                          ^~~~~
../Source/options.h:68:21: warning:   ‘devilution::string_view devilution::OptionEntryBase::name’ [-Wreorder]
   68 |         string_view name;
      |                     ^~~~
../Source/options.h:48:9: warning:   when initialized here [-Wreorder]
   48 |         OptionEntryBase(string_view key, OptionEntryFlags flags, string_view name, string_view description)
      |         ^~~~~~~~~~~~~~~
../Source/options.h: In constructor ‘devilution::OptionEntryBoolean::OptionEntryBoolean(devilution::string_view, devilution::OptionEntryFlags, devilution::string_view, devilution::string_view, bool)’:
../Source/options.h:95:14: warning: ‘devilution::OptionEntryBoolean::defaultValue’ will be initialized after [-Wreorder]
   95 |         bool defaultValue;
      |              ^~~~~~~~~~~~
../Source/options.h:94:14: warning:   ‘bool devilution::OptionEntryBoolean::value’ [-Wreorder]
   94 |         bool value;
      |              ^~~~~
../Source/options.h:79:9: warning:   when initialized here [-Wreorder]
   79 |         OptionEntryBoolean(string_view key, OptionEntryFlags flags, string_view name, string_view description, bool defaultValue)
      |         ^~~~~~~~~~~~~~~~~~
../Source/options.h: In constructor ‘devilution::OptionEntryEnumBase::OptionEntryEnumBase(devilution::string_view, devilution::OptionEntryFlags, devilution::string_view, devilution::string_view, int)’:
../Source/options.h:139:13: warning: ‘devilution::OptionEntryEnumBase::defaultValue’ will be initialized after [-Wreorder]
  139 |         int defaultValue;
      |             ^~~~~~~~~~~~
../Source/options.h:138:13: warning:   ‘int devilution::OptionEntryEnumBase::value’ [-Wreorder]
  138 |         int value;
      |             ^~~~~
../Source/options.h:126:9: warning:   when initialized here [-Wreorder]
  126 |         OptionEntryEnumBase(string_view key, OptionEntryFlags flags, string_view name, string_view description, int defaultValue)
```

--- a/Source/options.h
+++ b/Source/options.h
@@ -65,9 +65,9 @@ public:
 
 protected:
 	string_view key;
+	OptionEntryFlags flags;
 	string_view name;
 	string_view description;
-	OptionEntryFlags flags;
 	void NotifyValueChanged();
 
 private:
@@ -91,8 +91,8 @@ public:
 	void SaveToIni(string_view category) const override;
 
 private:
-	bool value;
 	bool defaultValue;
+	bool value;
 };
 
 class OptionEntryListBase : public OptionEntryBase {
@@ -135,8 +135,8 @@ protected:
 	void AddEntry(int value, string_view name);
 
 private:
-	int value;
 	int defaultValue;
+	int value;
 	std::vector<string_view> entryNames;
 	std::vector<int> entryValues;
 };

